### PR TITLE
Azurite service container for tests

### PR DESCRIPTION
**Pull request by bonsairobo**
Fixes https://github.com/ForesightMiningSoftwareCorporation/orica_libs/issues/1031

### CODE REVIEW

File: .github/workflows/rust-test.yml [hunk 1]
```diff
@@ -107,8 +111,18 @@ jobs:
           - type: miri
             toolchain: ${{ inputs.nightly_toolchain }}
       fail-fast: ${{ inputs.fail_fast == 'true' }}
+    services:
+      azurite:
+        # Hack for conditionally enabling the service:
+        # https://github.com/actions/runner/issues/822
+        image: ${{ (inputs.service_azurite) && 'mcr.microsoft.com/azure-storage/azurite' || '' }}
```
Comment (loispostula) [inline .github/workflows/rust-test.yml:118]: This is awesome

File: .github/workflows/rust-test.yml [hunk 2]
```diff
@@ -107,8 +111,18 @@ jobs:
...
+        ports:
+          - 10000:10000 # BLOB
+          - 10001:10001 # Queue
+          - 10002:10002 # Table
+        options: --health-cmd "nc -z 127.0.0.1 10000"
     steps:
-      - name: Set DATABASE_URL
+      - name: Start Postgres
```
Comment (loispostula) [inline .github/workflows/rust-test.yml:125]: Can you rewrite this part to just use the service like azurite?

**Review by loispostula — CHANGES_REQUESTED** [review]

File: .github/workflows/rust-test.yml (hunk 2 above)
Comment (bonsairobo) [reply .github/workflows/rust-test.yml:125]: Not really in scope for this PR, I think it should be a separate PR.

File: .github/workflows/rust-test.yml (hunk 1 above)
Comment (bonsairobo) [reply .github/workflows/rust-test.yml:118]: Would be more awesome as an "if" like jobs have.

**Review by loispostula — APPROVED** [review]

**Pull request merged**